metricbeat/mb/testing/data_generator.go

```diff
@@ -46,40 +47,58 @@ func WriteEvents(f mb.EventsFetcher, t *testing.T) error {
 	if len(events) == 0 {
 		return fmt.Errorf("no events were generated")
 	}
-	return createEvent(events[0], f)
+
+	fullEvent := CreateFullEvent(f, events[0])
+	WriteEventToDataJSON(t, fullEvent)
+	return nil
 }
 
-func createEvent(event common.MapStr, m mb.MetricSet) error {
-	path, err := os.Getwd()
+func CreateFullEvent(ms mb.MetricSet, metricSetData common.MapStr) common.MapStr {
+	startTime, err := time.Parse(time.RFC3339Nano, "2016-05-23T08:05:34.853Z")
 	if err != nil {
-		return err
+		panic(err)
 	}
 
-	startTime, _ := time.Parse(time.RFC3339Nano, "2016-05-23T08:05:34.853Z")
-
 	build := module.EventBuilder{
-		ModuleName:    m.Module().Name(),
-		MetricSetName: m.Name(),
-		Host:          m.Host(),
+		ModuleName:    ms.Module().Name(),
+		MetricSetName: ms.Name(),
+		Host:          ms.Host(),
 		StartTime:     startTime,
 		FetchDuration: 115 * time.Microsecond,
-		Event:         event,
+		Event:         metricSetData,
 	}
 
-	fullEvent, _ := build.Build()
+	fullEvent, err := build.Build()
+	if err != nil {
+		panic(err)
+	}
 	fullEvent["beat"] = common.MapStr{
 		"name":     "host.example.com",
 		"hostname": "host.example.com",
 	}
 
-	// Delete meta data as not needed for the event output here
-	delete(fullEvent, "_event_metadata")
+	// Delete meta data as not needed for the event output here.
+	delete(fullEvent, common.EventMetadataKey)
 
-	output, _ := json.MarshalIndent(fullEvent, "", "    ")
+	return fullEvent
+}
 
-	err = ioutil.WriteFile(path+"/_meta/data.json", output, 0644)
+func WriteEventToDataJSON(t testing.TB, fullEvent common.MapStr) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function WriteEventToDataJSON should have comment or be unexported